parquet-column/src/main/java/org/apache/parquet/schema/PrimitiveStringifier.java

```diff
@@ -0,0 +1,358 @@
+  }
+
+  /**
+   * @param value
+   *          the value to be stringified
+   * @return the string representation for {@code value}
+   * @throws UnsupportedOperationException
+   *           if value type is not supported by this stringifier
+   */
+  public String stringify(float value) {
+    throw new UnsupportedOperationException(
+        "stringify(float) was called on a non-float stringifier: " + toString());
+  }
+
+  /**
+   * @param value
+   *          the value to be stringified
+   * @return the string representation for {@code value}
+   * @throws UnsupportedOperationException
+   *           if value type is not supported by this stringifier
+   */
+  public String stringify(double value) {
+    throw new UnsupportedOperationException(
+        "stringify(double) was called on a non-double stringifier: " + toString());
+  }
+
+  /**
+   * @param value
+   *          the value to be stringified
+   * @return the string representation for {@code value}
+   * @throws UnsupportedOperationException
+   *           if value type is not supported by this stringifier
+   */
+  public String stringify(Binary value) {
+    throw new UnsupportedOperationException(
+        "stringify(Binary) was called on a non-Binary stringifier: " + toString());
+  }
+
+  private static final String BINARY_NULL = "null";
+  private static final String BINARY_HEXA_PREFIX = "0x";
+  private static final String BINARY_INVALID = "<INVALID>";
+
+  static abstract class BinaryStringifierBase extends PrimitiveStringifier {
+    private BinaryStringifierBase(String name) {
+      super(name);
+    }
+
+    @Override
+    public final String stringify(Binary value) {
+      return value == null ? BINARY_NULL : stringifyNotNull(value);
+    }
+
+    abstract String stringifyNotNull(Binary value);
+  }
+
+  static final PrimitiveStringifier DEFAULT_STRINGIFIER = new BinaryStringifierBase("DEFAULT_STRINGIFIER") {
+    private final char[] digits = { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F' };
+
+    @Override
+    public String stringify(boolean value) {
+      return Boolean.toString(value);
+    }
+
+    @Override
+    public String stringify(int value) {
+      return Integer.toString(value);
+    }
+
+    @Override
+    public String stringify(long value) {
+      return Long.toString(value);
+    }
+
+    @Override
+    public String stringify(float value) {
+      return Float.toString(value);
+    }
+
+    @Override
+    public String stringify(double value) {
+      return Double.toString(value);
+    }
+
+    @Override
+    String stringifyNotNull(Binary value) {
+      ByteBuffer buffer = value.toByteBuffer();
+      StringBuilder builder = new StringBuilder(2 + buffer.remaining() * 2);
+      builder.append(BINARY_HEXA_PREFIX);
+      for (int i = buffer.position(), n = buffer.limit(); i < n; ++i) {
+        byte b = buffer.get(i);
+        builder.append(digits[(b >>> 4) & 0x0F]);
```

Comment:
Have you considered using String.format("%02X", b)?

Reply:
The performance overhead for using `String.format` for each byte elements would be significant. A don't think the exchanging of the existing two lines (one for the upper and one for the lower bits) to one would worth it.